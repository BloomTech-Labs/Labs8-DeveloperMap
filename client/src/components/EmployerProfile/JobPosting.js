import React from 'react';
import styled from 'styled-components';
import Heart from '../../images/hollow-heart.png';
import FavHeart from '../../images/favorites-icon.png';

class JobPosting extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  componentDidMount() {
    // Axios.get().then(response => {
    //   const favoritedList = response.data;
    //   if (favoritedList.includes(this.props.post.jobId)) {
    //     document.querySelector(`#${jobId}`).setAttribute('src', FavHeart);
    //   }
    // });
    const favoritedList = ['jb1'];
    if (favoritedList.includes(this.props.post.jobId)) {
      document
        .querySelector(`#${this.props.post.jobId}`)
        .setAttribute('src', FavHeart);
    }
  }

  passPostThenToggle = e => {
    this.props.favToggle(e, this.props.post);
  };
  render() {
    return (
      <Job>
        <div>
          <h2>{this.props.post.companyName}</h2>
          <h3>{this.props.post.date}</h3>
          <h4>{this.props.post.jobLink}</h4>
          <h4>{this.props.post.jobTitle}</h4>
          <h4>{this.props.post.location}</h4>
        </div>
        <div>
          <img
            id={this.props.post.jobId}
            alt="hollow heart"
            src={Heart}
            style={{ width: '50px' }}
            onClick={this.passPostThenToggle}
          />
        </div>
      </Job>
    );
  }
}

const Job = styled.div`
  box-shadow: 0 4px 2px -2px gray;
  display: flex;
  justify-content: space-between;
  height: 130px;
  max-width: 350px;
  width: 100%;
  border: 0.7px solid rgba(220, 220, 220, 0.6);
  padding: 5%;
  margin-bottom: 15px;
  &:hover {
    box-shadow: none;
  }
`;

export default JobPosting;
