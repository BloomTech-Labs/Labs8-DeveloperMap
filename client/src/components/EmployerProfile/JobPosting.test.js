import React from 'react';
import renderer from 'react-test-renderer';
import JobPosting from './JobPosting';

describe(`job posting component`, () => {
    it(`should render without error`, () => {
        const props = {
            favoritedList: [{
                companyName: "Business",
                companyUid: "pIybk7iZOxYogp4GUTocjDnj9DH2",
                companyUrl: "/employer/pIybk7iZOxYogp4GUTocjDnj9DH2",
                date: "12/3/18",
                jobId: "-LSvdxaCKKJj55LBbgtK",
                jobLink: "indeed.com",
                jobTitle: "Com Sci Professional",
                location: {city: "Big Bear Lake", coordinates: [-116.88096, 34.243862], state: "CA", street: "42200 Moonridge Rd", zip: "92315"}}],
            post: {
                jobId: "uid2",
                jobLink: 'indeed.com'
            }
        }
        const tree = renderer.create(
            <JobPosting {...props}/>
          ).toJSON();
          expect(tree).toMatchSnapshot();
    });

    it(`should not break with no favorited list`, () => {
        const props = {
            favoritedList: [],
            post: {
                jobId: "uid2",
                jobLink: 'indeed.com'
            }
        }
        const tree = renderer.create(
            <JobPosting {...props}/>
          ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})