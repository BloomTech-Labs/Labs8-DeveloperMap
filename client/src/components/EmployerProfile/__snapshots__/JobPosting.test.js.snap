// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`job posting component should render without error 1`] = `
<div
  className="sc-bZQynM bRNLMT"
>
  <div>
    <h2 />
    <h3 />
    <h4>
      <a
        href="https://indeed.com"
      >
        indeed.com
      </a>
    </h4>
    <h4 />
  </div>
  <div>
    <img
      alt="hollow heart"
      id="uid2"
      onClick={[Function]}
      src="hollow-heart.png"
      style={
        Object {
          "width": "50px",
        }
      }
    />
  </div>
</div>
`;
